library/std/src/collections/hash/map.rs

```diff
@@ -973,6 +973,9 @@ where
     /// Returns an array of length `N` with the results of each query. For soundness, at most one
     /// mutable reference will be returned to any value. `None` will be used if the key is missing.
     ///
+    /// This method performs a check to ensure there are no duplicate keys, which has time-complexity O(n^2),
```

Comment:
Yes, $O(n \cdot \log n) \subset O(n^2)$, but I'd prefer this to be worded in such a way that we can improve the bound later.
```suggestion
    /// This method performs a check to ensure there are no duplicate keys, which currently has a time-complexity of O(n^2),
```